site/blog/tutorials/iasql-on-ec2.mdx

```diff
@@ -0,0 +1,184 @@
+
+:::note
+An IaSQL RPC is a function defined in Postgres that upon calling, sends a request to the IaSQL engine to run a pre-define logic and returns results to the Postgres console. In the above example [`key_pair_import`](https://iasql.com/docs/modules/aws/tables/aws_ec2_rpcs_import.KeyPairImportRpc/) is an IaSQL RPC that triggers the IaSQL engine to send a request to AWS for importing the public key.
+:::
+
+Now it's time to launch our EC2 instance. Before we go for the instance itself, let's first define the security group that's going to be attached to it. For this purpose, we'll need [`aws_security_group`](https://iasql.com/docs/modules/aws/aws_security_group/) module. But since `aws_ec2` module depends on it, it's already installed because IaSQL automatically installs the dependencies for modules upon an `iasql_install` call.
+
+<img src={"/img/module-dependency.png"} alt={"aws_ec2 module dependencies"} />
+
+```sql
+INSERT INTO security_group (description, group_name)
+VALUES ('IaSQL security group', 'iasql_sg');
+
+INSERT INTO security_group_rule (is_egress, ip_protocol, from_port, to_port, cidr_ipv4, description, security_group_id)
+SELECT t.is_egress,
+       t.ip_protocol,
+       t.from_port,
+       t.to_port,
+       t.cidr_ipv4::cidr,
+       t.description,
+       security_group.id
+FROM security_group,
+     (VALUES (FALSE, 'tcp', 22, 22, '0.0.0.0/0', 'ssh port is accessible from all ips'),
+             (FALSE, 'tcp', 9879, 9879, '<your-ip>/0', 'open dashboard to just me'),
+             (FALSE, 'tcp', 5432, 5432, '0.0.0.0/0', 'postgres port is accessible from all ips'),
+             (TRUE, 'tcp', 1, 65535, '0.0.0.0/0',
+              'allow outgoing traffic to anywhere')) AS t (is_egress, ip_protocol, from_port, to_port, cidr_ipv4, description)
+WHERE security_group.group_name = 'iasql_sg';
+```
+The first query simply creates a security group named `iasql_sg`. The second query seems a bit hard at the first sight, but it can be re-written as multiple `INSERT` queries:
+```sql
+INSERT INTO security_group_rule (is_egress, ip_protocol, from_port, to_port, cidr_ipv4, description, security_group_id)
+VALUES (FALSE, 'tcp', 22, 22, '0.0.0.0/0', 'ssh port is accessible from all ips',
+        (SELECT id FROM security_group WHERE group_name = 'iasql_sg'));
+```
+The reason we've written it that way is to avoid repeating the inner `SELECT id FROM security_group` query four times. The `security_group_id` column has an FK set to the `security_group` table's `id` column. Therefore, we need to insert the correct `id` for the security group we just created.
+
+Since the IaSQL dashboard does not have a default authorization mechanism set, remember to replace `<your-ip>` value in the above queries with your own IP address.
```

Comment:
maybe highlight this as a warning or important note?

Reply:
Sure.